### PR TITLE
Test with random buffer sizes

### PR DESCRIPTION
**Pull request by tranma**
! @amosr @jystic

### CODE REVIEW

File: icicle-compiler/test/Icicle/Test/Sea/Psv.hs [hunk 1]
```diff
@@ -174,16 +179,39 @@ failWithError' prints wt = \case
...
+    psv <- genPsvOpts wt
+    return $ WellTypedPsv wt psv
+
+genPsvOpts :: WellTyped -> Gen PsvOpts
+genPsvOpts wt = do
+  -- maximum number of rows to read before compute
+  let row x = x + 1
+  maxRowCount <- row . getPositive <$> arbitrary
+  -- the buffer needs to be at least as large as a single line
+  let str x = x + longestLine wt + 4
```
Comment (jystic) [inline icicle-compiler/test/Icicle/Test/Sea/Psv.hs]: do we ever want to test what happens when the buffer isn't long enough? /not sure

**Comment by jystic** [comment]
Looks good to me 👍

File: icicle-compiler/test/Icicle/Test/Sea/Psv.hs (hunk 1 above)
Comment (tranma) [inline icicle-compiler/test/Icicle/Test/Sea/Psv.hs]: I'm not sure what the sensible behaviour in that case is. Right now I believe ice/icicle-bench segfaults.

Comment (jystic) [inline icicle-compiler/test/Icicle/Test/Sea/Psv.hs]: Yeah we probably shouldn't do that :) I think either we should drop the row, or fail the whole job with an error. Dropping the row is probably the nicest, not sure how hard that is though.

Comment (amosr) [inline icicle-compiler/test/Icicle/Test/Sea/Psv.hs]: ice merge should blow up with a reasonable error message in this case, so I think it's adequate for now

Comment (tranma) [inline icicle-compiler/test/Icicle/Test/Sea/Psv.hs]: cool, will rebase and merge this

**Comment by jystic** [comment]
Bump, should be good to go 👍

**Pull request merged**